pages/[id].jsx

```diff
@@ -1,30 +1,14 @@
 import { useEffect, useState } from 'react';
-import { useRouter } from 'next/router';
 import Head from 'next/head';
 
 import apiService from '../services/apiService';
-import LoadingStatus, { loadingStates } from '../components/LoadingStatus';
 
-export default function DetailPage() {
+export default function DetailPage({ fetchedTeamMember }) {
   const [teamMemberDetails, setTeamMemberDetails] = useState({});
-  const [loadingStatus, setLoadingStatus] = useState(loadingStates.EMPTY);
-  const router = useRouter();
-  const { id } = router.query;
 
   useEffect(() => {
-    const getAndSetData = async () => {
-      try {
-        setLoadingStatus(loadingStates.LOADING);
-        const fetchedTeamMember = await apiService.getTeamMemberById('api/fellowship/', id);
-        setTeamMemberDetails(fetchedTeamMember);
-        setLoadingStatus(loadingStates.LOADED);
-      } catch (err) {
-        setLoadingStatus(loadingStates.ERROR);
-      }
-    };
-
-    if (id) getAndSetData();
-  }, [id]);
+    setTeamMemberDetails(fetchedTeamMember);
+  }, [fetchedTeamMember]);
```

Comment:
Since the data is being passed in as a prop, `useState` and `useEffect` are overkill for this. You should be able to delete these lines and then update all instances of `teamMemberDetails` with `fetchedTeamMember`.

That should have the added bonus of eliminating the millisecond of "Greetings !" that flashes on page load.

Reply:
Done and done!